src/bmpcodec.c

```diff
@@ -899,7 +897,6 @@ gdip_read_bmp_image (void *pointer, GpImage **image, ImageSource source)
 
 		result->active_bitmap->palette = GdipAlloc (sizeof(ColorPalette) + sizeof(ARGB) * palette_entries);
 		if (result->active_bitmap->palette == NULL) {
-			status = OutOfMemory;
```

Comment:
instead of removing `status` in this function I think we should return it in the `error:` label below instead of the hardcoded `InvalidParameter`. At least that seems to me what the original intention was and also what the other functions do :)

Reply:
that makes sense, fixed